### ISSUE TITLE
Commander Skills

### ISSUE THREAD

**Issue by Snayff** [Snayff/nqp2#84]
## Suggested Feature 
Skills are ways of the player influencing Combat; do damage, apply effects such as slow or stun etc.

## Requirements 
- [ ]  Commanders start with a Skill, known as their signature.
- [ ]  The player's commander can learn new skills, up to 5. 
    - [ ]  When full, new skills can replace old ones. 
    - [ ]  The signature skill cannot be replaced. 
- [ ]  Skills have a cooldown. This cooldown reduces with each new Room the player travels to. 
- [ ] Skills cost energy to use. 
- [ ]  When casting, the selected skill is indicated. 
- [ ]  When casting, the area of effect of the skill is indicated. 
- [ ]  When casting, there is a delay from the player confirming and the skill being cast. This is set per skill.
- [ ]  Skills use 1 of 3 types of targeting:
    - [ ]  Target the floor 
    - [ ]  Target a specific unit: [player, enemy] (and effect all the entities in the unit, regardless of actual position)
    - [ ]  Target global:[player, enemy] (affect all units of applicable side, e.g. player or enemy)

## Additional Context
![chrome_xsxLtl4m3Z](https://user-images.githubusercontent.com/23128268/129490130-9d3390b2-c7f2-4cfd-af2f-cc2f7bc9fba2.png)

**Comment by PuddingChef** [comment]
@Snayff I think we mentioned skills also having levels. If, at the end of the level, the player doesn't like the new skills on offer, they can level up an existing one

**Comment by Snayff** [comment]
Yep, you're right, but I think I can add that bit myself once the
foundation is in.

**Comment by Snayff** [comment]
@DaFluffyPotato please add a couple of the active and passive skills, along with the system, and I'll add the rest as separate issues afterwards.

**Comment by Snayff** [comment]
> I'm thinking Chain Lighting, Poison Cloud, Smoke Screen, Taunting Obelisk, Curse of Fear, Freeze, Howling Winds, and ~2 of the passives (just to make the system for it since they should be easy once there's a system).
Agreed with @DaFluffyPotato which subset of skills they will implement. I'll separate the rest out to their own issues.
